daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/SValue.scala

```diff
@@ -193,13 +193,71 @@ object SValue {
       discard[Int](`SMap Ordering`.compare(k, k))
     }
 
-    def apply(isTextMap: Boolean, entries: Iterator[(SValue, SValue)]): SMap = {
+    /** Build an SMap from an indexed sequence of SValue key/value pairs.
+      *
+      * SValue keys are assumed to be in ascending order - hence the SMap's TreeMap will be built in time O(n) using a
+      * sorted map specialisation.
+      */
+    def apply(
```

Comment:
I would prefer using a different name for the case we require the entries are ordered. 

It would be make the code using the method clearer.

```suggestion
    def fromOrderedEntries(
```